### PR TITLE
ECL-376: ECL Account - Payments Tile

### PR DESCRIPTION
**Pull request by nemanjatomanovic**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyaccount/controllers/AccountController.scala [hunk 1]
```diff
@@ -48,20 +49,31 @@ class AccountController @Inject() (
...
           auditAccountViewed(o)
-
           o match {
             case Some(obligationData) =>
-              Ok(
-                view(
-                  request.eclRegistrationReference,
-                  ViewUtils.formatLocalDate(registrationDate),
-                  getLatestObligation(obligationData)
+              financialDataService.latestFinancialObligation.map { financialData =>
```
Comment (TheDanielDoyle) [inline app/uk/gov/hmrc/economiccrimelevyaccount/controllers/AccountController.scala:56]: Consider this to cut down duplication

`def onPageLoad: Action[AnyContent] = authorise.async { implicit request =>
    enrolmentStoreProxyService.getEclRegistrationDate(request.eclRegistrationReference).flatMap { registrationDate =>
      obligationDataConnector
        .getObligationData()
        .flatMap { o =>
          auditAccountViewed(o)
          financialDataService.latestFinancialObligation.map { financialData =>
            Ok(
              view(
                request.eclRegistrationReference,
                ViewUtils.formatLocalDate(registrationDate),
                o match {
                  case Some(obligationData) => getLatestObligation(obligationData)
                  case None => None
                },
                financialData
              )
            )
          }
        }
    }
  }`

File: app/uk/gov/hmrc/economiccrimelevyaccount/models/FinancialDataResponse.scala [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      extends HttpReads[Either[FinancialDataErrorResponse, FinancialDataResponse]] {
+    override def read(
+      method: String,
+      url: String,
+      response: HttpResponse
+    ): Either[FinancialDataErrorResponse, FinancialDataResponse] =
+      response.status match {
+        case OK                    =>
+          response.json.validate[FinancialDataResponse] match {
+            case JsSuccess(response, _) => Right(response)
```
Comment (TheDanielDoyle) [inline app/uk/gov/hmrc/economiccrimelevyaccount/models/FinancialDataResponse.scala:35]: Need a JsError case here.

File: app/uk/gov/hmrc/economiccrimelevyaccount/models/FinancialDataResponse.scala [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+          }
+        case INTERNAL_SERVER_ERROR =>
+          response.json.validate[FinancialDataErrorResponse] match {
+            case JsSuccess(errorResponse, _) => Left(errorResponse)
```
Comment (TheDanielDoyle) [inline app/uk/gov/hmrc/economiccrimelevyaccount/models/FinancialDataResponse.scala:39]: Need a JsError case here.

**Review by TheDanielDoyle — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/economiccrimelevyaccount/controllers/AccountController.scala (hunk 1 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyaccount/controllers/AccountController.scala]: Nice catch, thanks :)

File: app/uk/gov/hmrc/economiccrimelevyaccount/models/FinancialDataResponse.scala (hunk 3 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyaccount/models/FinancialDataResponse.scala:46]: Added

File: app/uk/gov/hmrc/economiccrimelevyaccount/models/FinancialDataResponse.scala (hunk 2 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyaccount/models/FinancialDataResponse.scala:35]: Added

**Review by TheDanielDoyle — APPROVED** [review]

**Pull request merged**